### ISSUE TITLE
`ucal_getTimeZoneTransitionDate(,UCAL_TZ_TRANSITION_NEXT,,)` from last 64-bit representable transition doesn't step forward or fail

### ISSUE THREAD

**Issue by ediosyncratic** [unicode-org/icu4x#1742]
On Debian/testing Linux running with ICU4C version 67.1, asking for the (non-inclusive) next transition (in CET) when that would be more than (1<<63) milliseconds after the start of 1970, I found that the call to `ucal_getTimeZoneTransitionDate(,UCAL_TZ_TRANSITION_NEXT,,&status)` succeeded (`status` was still `U_ZERO_ERROR`, function returned true) rather than either giving me back a time Qt can't represent or failing (thereby telling me it can't report a later transition).
For contrast, `ucal_getTimeZoneTransitionDate(,UCAL_TZ_TRANSITION_PREVIOUS,,)` returns false for the earliest transition that a 64-bit signed int can represent in milliseconds before 1970, correctly communicating that it is unable to represent an earlier transition, or possibly that it knows there was no earlier transition (since this time-stamp is, of course, well before the invention of time-zones, so the TZ db assumes everyone was on local solar mean time).
The documentation of `UCAL_TZ_TRANSITION_NEXT`, particularly when read alongside that of `UCAL_TZ_TRANSITION_NEXT_INCLUSIVE`, is quite clear that, on success, the returned time-stamp should be strictly later than the input one.

Worked around in Qt by https://codereview.qt-project.org/c/qt/qtbase/+/402156 after being discovered by a `QDate` selftest when I built with ICU (and hacked `QTimeZone`'s constructors to use its ICU back-end instead of using of the TZ db).
The test that found it was `tst_QDate`'s `startOfDay_endOfDay_fixed:last-full` at 183882177219600000 ms after the start of 1970.

**Comment by ediosyncratic** [comment]
Minimal reproducer:
```
#include <unicode/ucal.h>
#include <inttypes.h>

int main(void) {
    UErrorCode status = U_ZERO_ERROR;
    UCalendar *ucal = ucal_open(u"Europe/Oslo", 11, "en", UCAL_GREGORIAN, &status);
    if (!ucal || !U_SUCCESS(status))
        return 2;

    int64_t last = 183882177219600000;
    ucal_setMillis(ucal, last, &status);
    double tranMSecs = 0.0;
    if (!ucal_getTimeZoneTransitionDate(ucal, UCAL_TZ_TRANSITION_NEXT, &tranMSecs, &status))
        return 0; // Admits defeat
    if (!U_SUCCESS(status))
        return 0; // At least it set status
    if (tranMSecs > last)
        return 0; // Correctly succeeded, returning a time later than last
    return 1; // False "success"
}
```

**Comment by ediosyncratic** [comment]
That Jira requires me to set up an Atlassian account. You'll miss out on reports due to folk not being willing to jump through that hoop.  However, not this one:
https://unicode-org.atlassian.net/browse/ICU-21975

The components listed in the form were not very intelligible, so I ended up guessing at which one to set.
Jira told me, as I created the issue, that it was not visible, but I see no way to control that.  It is unassigned.
